src/binary/uint.rs

```diff
@@ -26,23 +30,45 @@ impl DecodedUInt {
             ));
         }
 
-        // Create a stack-allocated buffer to hold the data we're going to read in.
-        let mut buffer = [0u8; MAX_UINT_SIZE_IN_BYTES];
+        if length <= UINT_STACK_BUFFER_SIZE {
+            let buffer = &mut [0u8; UINT_STACK_BUFFER_SIZE];
+            DecodedUInt::read_using_buffer(data_source, length, buffer)
+        } else {
+            // We're reading an enormous int. Heap-allocate a Vec to use as storage.
+            let mut buffer = vec![0u8; length];
+            DecodedUInt::read_using_buffer(data_source, length, buffer.as_mut_slice())
+        }
+    }
+
+    fn read_using_buffer<R: IonDataSource>(
```

Comment:
PR tour: this helper method was broken out so the core reading logic would work with either a stack- or heap-allocated buffer.